nucypher/policy/conditions/lingo.py

```diff
@@ -104,21 +106,44 @@ def __init__(self, comparator: str, value: Any):
 
         self.comparator = comparator
         self.value = value
+        self.key = key
 
     def _sanitize_value(self, value):
         try:
             return ast.literal_eval(str(value))
         except Exception:
             raise self.InvalidExpression(f'"{value}" is not a permitted value.')
 
+    def _process_data(self, data: Any) -> Any:
+        """
+        If a key is specified, return the value at that key in the data if data is a dict or list-like.
+        Otherwise, return the data.
+        """
+        if self.key and isinstance(data, dict):
+            try:
+                processed_data = data[self.key]
+            except KeyError:
+                raise KeyError(f"Key '{self.key}' not found in return data.")
+        elif isinstance(self.key, int) and isinstance(data, (list, tuple)):
+            try:
+                processed_data = data[self.key]
+            except IndexError:
+                raise IndexError(f"Index '{self.key}' not found in return data.")
```

Comment:
Should this instead be something like:
```python
if self.key:
    if isinstance(data, dict):
          ...
     elif instance(self.key, int) and isinstance(data, (list, tuple)):
          ...
     else:
           ... <some exception about incompatible data>
else:
    processed_data = data
```

Feels like we are missing the exception case for incompatible data i.e. the data is not a dict, list, or tuple.